src/modele/Customer.java

```diff
@@ -40,58 +34,81 @@
 @DiscriminatorValue("2")
 public class Customer extends Point implements Serializable {
 
-    private static final long serialVersionUID = 1L;
-    @Id
-    @GeneratedValue(strategy = GenerationType.AUTO)
-    private Long id;
     @Basic(optional = false)
     @Column(name = "POSITION")
     private Integer position;
+
     @OneToMany(cascade = CascadeType.ALL, mappedBy = "customer")
     private Set<Demand> customerDemands;
-
+    
+    /**
+     * No-argument constructor
+     */
     public Customer() {
         super();
         this.position = 0;
-        this.customerDemands = new HashSet<Demand>();
```

Comment:
why ?

Reply:
still in the code, I don't understand why it's in red